src/robusta/integrations/kubernetes/base_triggers.py

```diff
@@ -42,6 +45,79 @@ def get_event_name(self) -> str:
     def get_event_description(self) -> str:
         return f"{self.k8s_payload.operation}-{self.k8s_payload.kind}-{self.k8s_payload.apiVersion}"
 
+    def get_scope_matching_data(self):
+        if self.k8s_payload.obj and "metadata" in self.k8s_payload.obj:
+            meta = self.k8s_payload.obj["metadata"]
+        else:
+            return {}
+        data = {}
+        for meta_key in ["name", "namespace", "labels", "annotations"]:
+            if meta_key in meta:
+                data[meta_key] = meta[meta_key]
+        data["attributes"] = meta
+        return data
+
+    def scope_match_attributes(self, attr_matcher, attr_value):
+        # attr_value is the full payload (nested dicts/lists)
+        # attr_matcher is a comma-separated list of jsonpath_ng-compatible search path expressions
+        # (for example "spec.containers[*].name=bad, status.phase=Pending")
+        try:
+            return all(self.match_attr_by_path(expr, attr_value) for expr in attr_matcher.split(","))
+        except InvalidMatcher:
+            logging.error(f'invalid attribute-matching expression "{attr_matcher}')
+
+    def match_attr_by_path(self, expr, data):
+        try:
+            path, value = expr.rsplit("=", 1)
+        except ValueError:  # no "=" in the expression
+            raise InvalidMatcher
+        path = path.strip()
+        if path.endswith("!"):
+            path = path[:-1].strip()
+            expected_result = False
+        else:
+            expected_result = True
+        value = value.strip()
+        try:
+            expr = jsonpath_ng.parse(path)
+        except jsonpath_ng.exceptions.JSONPathError:
+            logging.error(f'invalid path expression "{path}"')
+            return not expected_result
```

Comment:
The semantics in this case is a matter of choice IMO. Obviously, `x.y=z` should return `False` in case there is no such key, but in the case of `!=` I see it as an arbitrary choice (both return values make some sense). We can go with `False` for both `=` and `!=`, no problem.